From 3efdca0462a34c8933b80f5fbe6640788d7fb735 Mon Sep 17 00:00:00 2001
From: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
Date: Wed, 24 Jul 2024 08:54:58 +0200
Subject: [PATCH] dts: android: use retram as permanent storage for debug

Activate by default oops/panic logger with
- console : log buffer reserved for kernel message
- pmsg : log buffer reserved for userspace messages (logcat)

Signed-off-by: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
---
 .../st/stm32mp257f-ev1-ca35tdcid-resmem-overlay.dtsi   | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem-overlay.dtsi b/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem-overlay.dtsi
index f7d276e928af7..fd0d436cc5d0f 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem-overlay.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1-ca35tdcid-resmem-overlay.dtsi
@@ -53,7 +53,15 @@ cm33_sram2: cm33-sram2@a060000 {
 		};
 
 		cm33_retram: cm33-retram@a080000 {
-			reg = <0x0 0xa080000 0x0 0x1f000>;
+			reg = <0x0 0xa080000 0x0 0x1b000>;
+			no-map;
+		};
+
+		ramoops_retram: ramoops@a09b000 {
+			compatible = "ramoops";
+			reg = <0x0 0xa09b000 0x0 0x4000>;
+			console-size = <0x1000>;
+			pmsg-size = <0x3000>;
 			no-map;
 		};
 
-- 
2.34.1

